### DISCUSSION TITLE
Job Retry Behaviour

### DISCUSSION THREAD

**Discussion by mal-drop** [timgit/pg-boss#372] [Q&A] [answered]
What is the correct way to retry jobs which fail? We have a configured job queue with retry options:

```typescript
export const RETRY_CONFIG = {
    retryLimit: 10,
    retryDelay: 10,
    retryBackoff: true,
};
``` 

```typescript
        await boss.send(CoreQueue.WEBFLOW_CHECKOUT, webflowJobData, RETRY_CONFIG);
```

However if this job throws an error, it just moves to a completed state as is not retried.  What should we be doing to have this job run again?

```typescript
    boss.work(CoreQueue.WEBFLOW_CHECKOUT, options, (job: any) => {
        webflowCheckout(job)
            .then(() => job.done())
            .catch(error => job.done(error));
    });
```

**Comment by timgit** [comment, marked as answer]
Try using an async job handler to simplify job completion handling. You shouldn't normally need to explicitly call job.done(), as this was an older api used before async/await were added.
